src/Admin/AdminConsole/Controllers/ProvidersController.cs

```diff
@@ -275,4 +277,76 @@ public async Task<IActionResult> CreateOrganization(Guid providerId, Organizatio
 
         return RedirectToAction("Edit", "Providers", new { id = providerId });
     }
+
+    [HttpPost]
+    [SelfHosted(NotSelfHostedOnly = true)]
+    [RequirePermission(Permission.Provider_Edit)]
+    public async Task<IActionResult> Delete(Guid id, string providerName)
+    {
+        var providerOrganizations = await _providerOrganizationRepository.GetManyDetailsByProviderAsync(id);
+
+        if (providerOrganizations.Count > 0)
+        {
+            return BadRequest("You must unlink all clients before you can delete a provider");
+        }
+
+        var provider = await _providerRepository.GetByIdAsync(id);
+
+        var validationError = ValidateProviderName(provider, providerName);
+        if (!string.IsNullOrEmpty(validationError))
+        {
+            return BadRequest(validationError);
+        }
+
+        await _providerService.DeleteAsync(provider);
+        return NoContent();
+    }
+
+    [HttpPost]
+    [SelfHosted(NotSelfHostedOnly = true)]
+    [RequirePermission(Permission.Provider_Edit)]
+    public async Task<IActionResult> DeleteInitiation(Guid id, string providerEmail)
+    {
+        if (!IsValidEmail(providerEmail))
+        {
+            return BadRequest("Invalid provider admin email");
+        }
+
+        var provider = await _providerRepository.GetByIdAsync(id);
+        if (provider != null)
+        {
+            try
+            {
+                await _providerService.InitiateDeleteAsync(provider, providerEmail);
+            }
+            catch (BadRequestException ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+        return NoContent();
+    }
+
+    private static bool IsValidEmail(string email)
+    {
+        const string pattern = @"^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$";
+        var regex = new Regex(pattern);
+        return regex.IsMatch(email);
+    }
+
+    private string ValidateProviderName(Provider provider, string providerName)
```

Comment:
This method is only used once. I suggest moving this logic into the controller as it also checks if the Provider is null

Reply:
resolved